pkgs/by-name/ou/outfly/package.nix

```diff
@@ -0,0 +1,98 @@
+{
+  lib,
+  fetchFromGitea,
+  buildType ? "release",
+  platform ? "linux",
+  enableWayland ? true,
+  enableXorg ? true,
+  enableOpenGL ? true,
+  enableVulkan ? true,
+  rustPlatform,
+  makeDesktopItem,
+  pkg-config,
+  libxkbcommon,
+  alsa-lib,
+  libGL,
+  vulkan-loader,
+  wayland,
+  libXrandr,
+  libXcursor,
+  libX11,
+  libXi,
+  ...
+}:
+
+assert (platform == "linux") -> (enableWayland || enableXorg);
+assert (platform == "linux") -> (enableVulkan || enableOpenGL);
+
+rustPlatform.buildRustPackage rec {
+  pname = "outfly";
+  inherit buildType;
+  version = "0.13.0";
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "outfly";
+    repo = "outfly";
+    rev = "v${version}";
+    hash = "sha256-A43lLG5K9mEhRsv+paQa1so4Cyk75vyocAyU3k6wDiQ=";
+  };
+
+  buildNoDefaultFeatures = true;
+  buildFeatures = [
+    "embed_assets"
+  ] ++ lib.optionals enableXorg [ "x11" ] ++ lib.optionals enableWayland [ "wayland" ];
+
+  runtimeInputs =
+    [ libxkbcommon ]
+    ++ lib.optionals enableOpenGL [ libGL ]
+    ++ lib.optionals enableXorg [
+      libXrandr
+      libX11
+    ]
+    ++ lib.optionals enableVulkan [ vulkan-loader ];
+
+  buildInputs = [
+    alsa-lib.dev
+    libXcursor
+    libXi
+  ] ++ lib.optionals enableWayland [ wayland ];
+
+  nativeBuildInputs = [ pkg-config ];
+  doCheck = false;
+
+  postFixup = ''
+    patchelf $out/bin/outfly \
+    --add-rpath ${lib.makeLibraryPath runtimeInputs}
+  '';
+
+  cargoHash = "sha256-6qd52F5ub1aFxJHeFEysy6ZOOuqmcsCiGHJNPp6pwhA=";
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "outfly";
+      exec = "outfly";
+
+      desktopName = "OutFly";
+      categories = [ "Game" ];
+    })
+  ];
+  meta = {
+    description = "A breathtaking 3D space game in the rings of Jupiter";
+    homepage = "https://yunicode.itch.io/outfly";
+    downloadPage = "https://codeberg.org/outfly/outfly/releases";
+    changelog = "https://codeberg.org/outfly/outfly/releases/tag/v${version}";
+    license = with lib.licenses; [
+      cc-by-30
+      cc-by-40
+      cc-by-sa-20
+      cc-by-sa-30
+      cc0
+      gpl3
+      ofl
+      publicDomain
+    ];
+    platforms = lib.platforms.linux;
```

Comment:
There is theoretical support for macOS, but i dont currently have access to a mac, so if i added the macOS dependencies i would not be confident that it would run.
FreeBSD could work fine in theory, but *BSD are not platforms that are mentioned as supported or tested.
I dont want to advertise support for platforms that are likely broken, but if them breaking isnt a blocker, setting platforms to unix would be an option.

Reply:
I also have never tested it on FreeBSD or MacOS

Reply:
in that case i would prefer leaving it linux-only for now